src/ExtensionGallery/Startup.cs

```diff
@@ -1,109 +1,63 @@
 ﻿using System;
-using Microsoft.AspNet.Http;
 using System.Linq;
-using Microsoft.AspNet.Builder;
-using Microsoft.AspNet.Diagnostics;
-using Microsoft.AspNet.Hosting;
-using Microsoft.AspNet.Routing;
-using Microsoft.AspNet.StaticFiles;
-using Microsoft.Framework.Configuration;
-using Microsoft.Framework.DependencyInjection;
-using Microsoft.Framework.Logging;
-using Microsoft.Framework.Runtime;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.PlatformAbstractions;
+using Newtonsoft.Json.Serialization;
 
 namespace ExtensionGallery
 {
 	public class Startup
 	{
-		public Startup(IHostingEnvironment env, IApplicationEnvironment appEnv)
+		public Startup(IHostingEnvironment env)
 		{
             // Setup configuration sources.
-            Configuration = new ConfigurationBuilder(appEnv.ApplicationBasePath)
-				.AddJsonFile("config.json")
-				.AddEnvironmentVariables().Build();
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+            Configuration = builder.Build();
 		}
 
 		public IConfiguration Configuration { get; set; }
 
 		// This method gets called by the runtime.
 		public void ConfigureServices(IServiceCollection services)
 		{
-			// Add EF services to the services container.
-			//services.AddEntityFramework(Configuration)
-			//    .AddSqlServer()
-			//    .AddDbContext<ApplicationDbContext>();
-
-			//// Add Identity services to the services container.
-			//services.AddDefaultIdentity<ApplicationDbContext, ApplicationUser, IdentityRole>(Configuration);
-
 			// Add MVC services to the services container.
-			services.AddMvc();
+			services.AddMvc()
+                .AddJsonOptions(options => options.SerializerSettings.ContractResolver = new DefaultContractResolver());
 
-			// Uncomment the following line to add Web API servcies which makes it easier to port Web API 2 controllers.
-			// You need to add Microsoft.AspNet.Mvc.WebApiCompatShim package to project.json
-			// services.AddWebApiConventions();
-
-		}
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
```

Comment:
The IHttpContextAccessor service is not registered by default, per https://github.com/aspnet/Hosting/issues/793